### ISSUE TITLE
Easy Trunk branch change

### ISSUE THREAD

**Issue by gitbutler-client** [gitbutlerapp/gitbutler#2507]
I would like to change the trunk branch without the need to re-import the project

via @koppor

https://discord.com/channels/1060193121130000425/1176875943873232947/1193874541164908544

**Comment by guyzmo** [comment]
ok, I did not know that was possible and I've tried to do my changes with multiple vbranches together, but now I'm having a hard time reconciling my commits 😖

**Comment by monaye** [comment]
@schacon  what is the work-around ?

**Comment by MHHenriksen** [comment]
> @schacon what is the work-around ?

I assume it's re-importing the project.

I can point out a real world scenario where this feature would be valuable: Maintenance branches. I work with broadcast software that gets installed on-premise for each customer, and our customers are very change-averse. Any major update has considerable risk and cost just from bureaucracy, which leads to customers staying on known working versions as long as they can, and we have to maintain these versions for long periods. Thus we have one branch for each major version still under support in addition to a main branch (e.g. v7.2, v8.0, v8.1 and main). 

If I pick main as the trunk branch, I can't do anything with the maintenance branches because GitButler automatically merges the trunk into any applied branch, which we obviously don't want.

For certain internal reasons, we do not merge forward between these branches (otherwise I could just have selected the oldest maintained version as the GitButler Trunk and "apply" the other branches in order). 

So my current options are either the aforementioned workaround of re-importing the project every time I need to target a different release, or have one clone per version (wasting a lot of disk space and time replicating changes to each copy since the project is quite large).

**Comment by koppor** [comment]
@MHHenriksen I fully agree. I would also like to have gitbutler working in parallel to "normal" branches. I am thinking of `gh pr checkout 999`, which checks out a pull request by a contributor. After that, GitButler complains

> or have one clone per version (wasting a lot of disk space and time replicating changes to each copy since the project is quite large).

There is the cool [worktree feature](https://github.blog/2015-07-29-git-2-5-including-multiple-worktrees-and-triangular-workflows/) of git. A truly hidden gem. You can add more than one checkout. I don't know if GitBulter works with this, but I wanted to share, maybe its useful for you also in other contexts... -- Second thing: One can also clone locally. When doing on Linux, there is not any more disk space, because symbolic links are used (aren't they?).

**Comment by MHHenriksen** [comment]
@koppor Thanks for the tips! I did know about worktrees and have used them, which helps, but haven't actually tried it with GitButler. But I suspect the virtual branches might collide, since there's only one shared .git folder. And alas, these are Windows projects.

**Comment by koppor** [comment]
Another motivation to have GitButler supporting multiple branches: `gh pr checkout #`: When pull requests of external contributors need to be checked, these branches need a check out for "proper" try out and send review comments.

**Comment by Caleb-T-Owens** [comment]
<img width="1212" alt="Screenshot 2024-08-15 at 14 17 39" src="https://github.com/user-attachments/assets/d65ad75a-60cb-444e-ade3-8e853f469f8b">

We now have an experimental setting which you can use to change the trunk branch whist you have no branches applied.

@krlvi should we close this issue, or do you want to wait till we have moved it out of experimental mode?

**Comment by danmacode** [comment]
> <img alt="Screenshot 2024-08-15 at 14 17 39" width="1212" src="https://private-user-images.githubusercontent.com/13354155/358215524-d65ad75a-60cb-444e-ade3-8e853f469f8b.png?jwt=eyJ0eXAiOiJKV1QiLCJhbGciOiJIUzI1NiJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3NTYyMTI5ODksIm5iZiI6MTc1NjIxMjY4OSwicGF0aCI6Ii8xMzM1NDE1NS8zNTgyMTU1MjQtZDY1YWQ3NWEtNjBjYi00NDRlLWFkZTMtOGU4NTNmNDY5ZjhiLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNTA4MjYlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjUwODI2VDEyNTEyOVomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPWMxNTU5Y2RjZmMwOGFiYWE2OWZhOWExYTI5ODAwYmUyMjNhZWY0NzRkZjk3ZGI0ZjNkNmRmNzhiODllNmY5OTUmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0In0.Bm_gKOqA-gIxTmzpaKPoaYNMWTOyUVvG0pbXnpjBMw8">
> We now have an experimental setting which you can use to change the trunk branch whist you have no branches applied.
> 
> [@krlvi](https://github.com/krlvi) should we close this issue, or do you want to wait till we have moved it out of experimental mode?

@Caleb-T-Owens is this currently macOS-only? I don't see it on WSL or Windows.

I'll just try to create a worktree per target for now.

**Comment by Caleb-T-Owens** [comment]
<img width="1548" height="643" alt="Image" src="https://github.com/user-attachments/assets/d193cfc0-cfce-4d1a-85aa-1e1034716fc0" />

Hey! This is now enabled by default. You can change it in the project settings.

**Comment by Caleb-T-Owens** [comment]
Since this has been released for a while, I'm going to close the issue
